### PR TITLE
feat(core): add a new CardsTable component

### PR DESCRIPTION
**Pull request by saylikadam**
affects: @medly-components/core, @medly-components/theme

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Application / infrastructure changes
-   [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: packages/core/src/components/CardsTable/CardTableBody/Cell/Cell.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            isRowClickDisabled,
+        } = props
+    const formattedCell = useCallback(() => {
+            switch (column.formatter) {
+                case 'boolean':
+                    return <Text> {data} </Text>;
+                case 'react-component': {
+                    const Component = column.component;
+                    return (
+                        <div ref={childRef}>
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/CardTableBody/Cell/Cell.tsx:21]: Remove this Div

File: packages/core/src/components/CardsTable/CardTableBody/Row/Row.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const Row = styled(Card)<Props>`
+    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'grab' : 'inherit')};
+    box-shadow: none;
+    box-sizing: border-box;
+    border-radius: 0.8rem;
+    border: ${({ withLightTheme }) => withLightTheme ? 'default' : 'none'};
+
+    &:hover {
+        box-shadow: ${({ withLightTheme, theme }) => withLightTheme ? `0px 2px 4px ${theme.colors.grey[400]}` : `0px 4px 8px ${theme.colors.grey[400]}`};
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/CardTableBody/Row/Row.styled.tsx:13]: Change px to rem

File: packages/core/src/components/CardsTable/CardTableBody/Cell/Cell.tsx (hunk 1 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/CardTableBody/Cell/Cell.tsx]: Ok

File: packages/core/src/components/CardsTable/CardTableBody/Row/Row.styled.tsx (hunk 2 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/CardTableBody/Row/Row.styled.tsx]: ok

File: packages/core/src/components/CardsTable/Body/Body.test.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            data= {data}
+            columns={columns}
+            onRowClick={onRowClick}
+            rowClickDisableKey="disable"
+            uniqueKeyName="124"
+
+        />
+    );
+
+    it('should render cart table body properly', () => {
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/Body/Body.test.tsx:32]: spelling mistake

File: packages/core/src/components/CardsTable/CardsTable.test.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { render } from '@test-utils';
+import React from 'react';
+import { CardsTable } from './CardsTable'
+import { ColumnConfig } from './types';
+
+describe('CardsTable', () => {
+    it('should render CardsTable of cards table properly', () => {
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/CardsTable.test.tsx:7]: please correct the sentence

File: packages/core/src/components/CardsTable/docs/CardsTable.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                onSort={handleFilterData}
+                columns={columns}
+                onRowClick={() => null}
+            />
+        </>
+    )
+};
+
+
+export const DisabledRow = () => {
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/docs/CardsTable.stories.tsx:36]: As discussed we should remove this

File: packages/core/src/components/CardsTable/types.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    onSort?: (field: string, order: SortOrder) => void;
+    /** Function to be called on row click */
+    onRowClick?: (rowData: object) => void;
+    /** Key name to disable row click */
+    rowClickDisableKey?: string;
+    /** Key name to select background color */
+    withLightTheme?: boolean;
+
+}
+export interface StaticProps {
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/types.ts:56]: Remove this. This is specifically for table component.

File: packages/core/src/components/CardsTable/Body/Row/Row.styled.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    &:hover {
+        box-shadow: ${({ withLightTheme }) => withLightTheme ? '0 0.2rem 0.4rem rgba(176,188,200,0.60)' : '0 0.4rem 0.8rem rgba(176,188,200,0.60)'};
+    };
+
+    & > * {
+        background-color: ${({ theme, disabled }) => `${disabled ? theme.table.rowDisabledColor : 'inherit'}`};
+    };
+
+    :nth-child(2n) {
+        background: ${({ theme, withLightTheme}) => withLightTheme ? `${theme.colors.grey[50]}` : theme.table.rowBackgroundColor};
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/Body/Row/Row.styled.tsx:21]: we should get this property from cardTable theme

File: packages/core/src/components/CardsTable/Head/HeadCell/HeadCell.styled.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+const getTextStyle = ({ theme }: Props) => {
+    const {variants, defaults, weights} = theme.font
+
+    return css`
+        color: ${defaults.color};
+        text-transform: uppercase;
+        font-size: ${variants.h5.fontSize};
+        letter-spacing: ${variants.h5.letterSpacing};
+        font-weight: ${weights[variants.h5.fontWeight]};
+        display: flex;
```
Comment (gmukul01) [inline packages/core/src/components/CardsTable/Head/HeadCell/HeadCell.styled.tsx:17]: Why we need display flex over here?

File: packages/core/src/components/CardsTable/Body/Body.test.tsx (hunk 3 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/Body/Body.test.tsx:32]: ok

File: packages/core/src/components/CardsTable/CardsTable.test.tsx (hunk 4 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/CardsTable.test.tsx:7]: ok

File: packages/core/src/components/CardsTable/docs/CardsTable.stories.tsx (hunk 5 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/docs/CardsTable.stories.tsx:36]: its removed from .mdx but will also remove it from here

File: packages/core/src/components/CardsTable/types.ts (hunk 6 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/types.ts:56]: ok

File: packages/core/src/components/CardsTable/Body/Row/Row.styled.tsx (hunk 7 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/Body/Row/Row.styled.tsx:21]: got it

File: packages/core/src/components/CardsTable/Head/HeadCell/HeadCell.styled.tsx (hunk 8 above)
Comment (saylikadam) [reply packages/core/src/components/CardsTable/Head/HeadCell/HeadCell.styled.tsx]: removing

**Pull request merged**